tests/topotests/kernel_bypass_zebra_ecmp/test_kernel_bypass_ecmp.py

```diff
@@ -0,0 +1,168 @@
+    # ... and here it calls Mininet initialization functions.
+    tgen.start_topology()
+    print("topology started")
+    # This is a sample of configuration loading.
+    router_list = tgen.routers()
+    
+    # For all registred routers, load the zebra configuration file
+    for i, (rname, router) in enumerate(router_list.items(), 1):
+        router.load_config(
+            TopoRouter.RD_ZEBRA,
+            os.path.join(CWD, '{}/zebra.conf'.format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_BGP,
+            os.path.join(CWD, '{}/bgpd.conf'.format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_STATIC,
+            os.path.join(CWD, '{}/staticd.conf'.format(rname))
+        )
+    
+    # After loading the configurations, this function loads configured daemons.
+    tgen.start_router()
+
+
+    # Verify that we are using the proper version and that the BFD
+    # daemon exists.
+    for router in router_list.values():
+        # Check for Version
+        if router.has_version('<', '5.1'):
+            tgen.set_error('Unsupported FRR version')
+            break
+
+def teardown_module(mod):
+    "Teardown the pytest environment"
+    tgen = get_topogen()
+    # This function tears down the whole topology.
+    tgen.stop_topology()
+
+
+def test_ebgp_peers():
+    "Assert that BGP peers."
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    logger.info('waiting for bgp peers to go up')
+
+    for router in tgen.routers().values():
+        ref_file = '{}/{}/peers.json'.format(CWD, router.name)
+        expected = json.loads(open(ref_file).read())
+        test_func = partial(topotest.router_json_cmp,
+                            router, 'show bgp neighbors json', expected)
+        _, res = topotest.run_and_expect(test_func, None, count=10, wait=10.0)
+        assertmsg = '{}: bgp did not established'.format(router.name)
+        assert res is None, assertmsg
+
+def test_ebgp_convergence():
+    "Assert that BGP is converging."
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    logger.info('waiting for bgp peers to go up')
+
+    for router in tgen.routers().values():
+        ref_file = '{}/{}/ip_route_summary.json'.format(CWD, router.name)
+        expected = json.loads(open(ref_file).read())
+        test_func = partial(topotest.router_json_cmp,
+                            router, 'show ip route summary json', expected)
+        _, res = topotest.run_and_expect(test_func, None, count=10, wait=5)
+        assertmsg = '{}: bgp did not converge'.format(router.name)
+        assert res is None, assertmsg
+
+
+def test_kernel_route():
+    "Assert kernel route."
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    r1 = tgen.gears['r1']
+    test_func = partial(r1.cmd,"route -n | wc -l")
```

Comment:
Please use `ip route` instead. Also, it would be easier to check with `ip -j route`.

Reply:
ip -j route is not supported in low kernel version, like my testbed of ubuntu 18.04~4.15.0-193-generic.